### PR TITLE
Results

### PR DESCRIPTION
**Pull request by alex-jonas**
I am looking forward to a review of my results page (frontend, backend). Please ignore the comments in the code, sorry.

### CODE REVIEW

File: backend/src/Controller/GeocodeController.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                $content,
+                JsonResponse::HTTP_OK,
+                [],
+                true
+            );
+        }
+        return new JsonResponse(
+            ['success' => 'no'],
+            JsonResponse::HTTP_NOT_FOUND,
+            []
```
Comment (tine-krueger) [inline backend/src/Controller/GeocodeController.php:50]: Maybe the positive and user-wanted return should be the final one

File: backend/src/Controller/TrackController.php [hunk 2]
```diff
@@ -47,4 +48,42 @@ public function create(
...
+        foreach ($tracks as $track) {
+
+            $latDestination = $track->getFirstLat();
+            $lngDestination = $track->getFirstLon();
+            $calculatedDistance = $geodistance->getDistanceMeters($latOrigin, $lngOrigin, $latDestination, $lngDestination);
+            $track->setDistance($calculatedDistance);
+        }
+
+        usort($tracks, fn ($a, $b) => ($a->distance < $b->distance) ? -1 : 1);
+
```
Comment (tine-krueger) [inline backend/src/Controller/TrackController.php:81]: nice, think about refactoring those business logic to utils

File: frontend/src/components/Map.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      {tracks.map(({ firstLat, firstLon }, index) => (
+        <Marker key={index} position={{ lat: +firstLat, lng: +firstLon }} />
+      ))}
+    </GoogleMap>
+  )
+
+  return isLoaded ? map : null
+}
+/*
+
```
Comment (tine-krueger) [inline frontend/src/components/Map.js:64]: You still need that?

File: frontend/src/pages/Results.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        return false
+      } else if (
+        key === 'certYear' &&
+        Boolean(track[key]) !== filterCriteria[key]
+      ) {
+        return false
+      }
+    }
+    return true
+  })
```
Comment (tine-krueger) [inline frontend/src/pages/Results.js:59]: If it really has to be in FE, you should refactor

File: backend/src/Controller/GeocodeController.php (hunk 1 above)
Comment (alex-jonas) [reply backend/src/Controller/GeocodeController.php:50]: You mean the first one, right? I would consider this improvement to be rather dispensable.

File: backend/src/Controller/TrackController.php (hunk 2 above)
Comment (alex-jonas) [reply backend/src/Controller/TrackController.php:81]: ...noted on the refactoring todo list.

File: frontend/src/components/Map.js (hunk 3 above)
Comment (alex-jonas) [reply frontend/src/components/Map.js:64]: Not really. Well seen!

File: frontend/src/pages/Results.js (hunk 4 above)
Comment (alex-jonas) [reply frontend/src/pages/Results.js:59]: In a later branch more filter logic should happen in the backend. Thanks for this impulse.

**Pull request merged**